app/views/file.js

```diff
@@ -1357,31 +1360,14 @@ module.exports = Backbone.View.extend({
     return _.compact([dir, fileName]).join('/');
   },
 
-  upload: function(e, file, content, path) {
+  upload: function(e, file, content, path, successCb, failureCb) {
     // Loading State
     this.updateSaveState(t('actions.upload.uploading', { file: file.name }), 'saving');
 
     var uploadPath = path || this.defaultUploadPath(file.name);
     this.collection.upload(file, content, uploadPath, {
-      success: (function(model, res, options) {
-        var name = res.content.name;
-        var path = '{{site.baseurl}}/' + res.content.path;
-
-        // Take the alt text from the insert image box on the toolbar
-        var $alt = $('input[name="alt"]');
-        var value = $alt.val();
-        var image = (value) ?
-          '![' + value + '](' + path + ')' :
-          '![' + name + '](' + path + ')';
-
-        this.editor.focus();
-        this.editor.replaceSelection(image + '\n', 'end');
-        this.updateSaveState('Saved', 'saved', true);
-      }).bind(this),
-      error: (function(model, xhr, options) {
-        var message = util.xhrErrorMessage(xhr);
-        this.updateSaveState(message, 'error');
-      }).bind(this)
+      success: (successCb).bind(this),
+      error: (failureCb).bind(this)
```

Comment:
The `error` callbacks are all the same I think, so that can probably just stay put?